test/unit/image-optimizer/match-remote-pattern.test.ts

```diff
@@ -0,0 +1,235 @@
+    expect(m(p, new URL('https://example.com.uk'))).toBe(false)
+    expect(m(p, new URL('https://sub.example.com.uk'))).toBe(false)
+    expect(m(p, new URL('https://avatars.example.com'))).toBe(true)
+    expect(m(p, new URL('https://avatars.sfo1.example.com'))).toBe(true)
+    expect(m(p, new URL('https://avatars.iad1.example.com'))).toBe(true)
+    expect(m(p, new URL('https://more.avatars.iad1.example.com'))).toBe(true)
+  })
+
+  it('should match pathname pattern with single asterisk', () => {
+    const p = {
+      hostname: 'example.com',
+      pathname: '/act123/*/pic.jpg',
+    } as const
+    expect(m(p, new URL('https://com'))).toBe(false)
+    expect(m(p, new URL('https://example.com'))).toBe(false)
+    expect(m(p, new URL('https://sub.example.com'))).toBe(false)
+    expect(m(p, new URL('https://example.com.uk'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123/usr4'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123/usr4/pic'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123/usr4/picsjpg'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123/usr4/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr5/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr6/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/team/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act456/team/pic.jpg'))).toBe(false)
+    expect(m(p, new URL('https://example.com/team/pic.jpg'))).toBe(false)
+  })
+
+  it('should match pathname pattern with double asterisk', () => {
+    const p = { hostname: 'example.com', pathname: '/act123/**' } as const
+    expect(m(p, new URL('https://com'))).toBe(false)
+    expect(m(p, new URL('https://example.com'))).toBe(false)
+    expect(m(p, new URL('https://sub.example.com'))).toBe(false)
+    expect(m(p, new URL('https://example.com.uk'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123'))).toBe(false)
+    expect(m(p, new URL('https://example.com/act123/usr4'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr4/pic'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr4/picsjpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr4/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr5/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/usr6/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act123/team/pic.jpg'))).toBe(true)
+    expect(m(p, new URL('https://example.com/act456/team/pic.jpg'))).toBe(false)
+    expect(m(p, new URL('https://example.com/team/pic.jpg'))).toBe(false)
+  })
+
+  it('should throw when hostname is missing', () => {
+    const p = { protocol: 'https' } as const
+    expect(() => m(p, new URL('https://example.com'))).toThrow(
```

Comment:
```suggestion
    // @ts-ignore testing invalid input
    expect(() => m(p, new URL('https://example.com'))).toThrow(
```
Looks like we might need a `@ts-ignore` here since the type doesn't like testing the missing `hostname`